### PR TITLE
Improve super prediction

### PR DESCRIPTION
**Pull request by def-**
As suggested by cheeser0613

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Review by edg-l — APPROVED** [review]
it looks good for me but better if another also takes look, could have potential edge cases that i overlooked maybe

File: src/game/teamscore.cpp [hunk 1]
```diff
@@ -19,7 +20,7 @@ int CTeamsCore::Team(int ClientID) const
 
 void CTeamsCore::Team(int ClientID, int Team)
 {
-	m_Team[ClientID] = Team;
+	m_Team[ClientID] = clamp(Team, (int)TEAM_FLOCK, (int)TEAM_SUPER);
```
Comment (heinrich5991) [inline src/game/teamscore.cpp:23]: Haven't looked at the rest, but this stood out to me: This seems to hide bugs. Can you add a `dbg_assert` instead?

Comment (def-) [reply src/game/teamscore.cpp:23]: Done.

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**